data/src/main/java/org/oppia/data/persistence/PersistentCacheStore.kt

```diff
@@ -46,6 +46,12 @@ class PersistentCacheStore<T : MessageLite> private constructor(
   @GuardedBy("failureLock") private var deferredLoadCacheFailure: Throwable? = null
   private val cache = cacheFactory.create(CachePayload(state = CacheState.UNLOADED, value = initialValue))
 
+  init {
+    cache.observeChanges {
+      asyncDataSubscriptionManager.notifyChange(providerId)
```

Comment:
Why do we have to start observing in the init block itself?

Reply:
There doesn't seem to be another time that's good to do this, and it's effectively initialization work. We could lazily handle this at the time it's needed, but that adds a bit of additional complexity that doesn't have a clear win to me other than maybe thinning out the constructor (which is generally valuable for injected constructors, but this one isn't--it's only executed when a cache store is needed).

Reply:
Okay that one makes sense.